osi_environment.proto

```diff
@@ -324,4 +336,163 @@ message EnvironmentalConditions
+        //
+        // \par References:
+        // ISO 15469:2004(E) / CIE S 011/E:2003 : Spatial distribution of skylight
+        // https://en.wikipedia.org/wiki/Cloud_cover
+        // [1] https://www.eoas.ubc.ca/courses/atsc113/flying/met_concepts/01-met_concepts/01c-cloud_coverage/index.html
+        // [2] https://www.dwd.de/DE/service/lexikon/Functions/glossar.html?lv2=100932&lv3=101016
+        //
+        enum CloudLayerLevel
+        {
+            // Cloud layer level is unknown (must not be used in ground truth).
+            //
+            CLOUD_LAYER_LEVEL_UNKNOWN = 0;
+
+            // Other (unspecified but known) cloud layer level.
+            //
+            CLOUD_LAYER_LEVEL_OTHER = 1;
+
+            // 0/8 of the sky is covered with clouds.
+            //
+            CLOUD_LAYER_LEVEL_0 = 2;
+
+            // 1/8 of the sky is covered with clouds.
+            //
+            CLOUD_LAYER_LEVEL_1 = 3;
+
+            // 2/8 of the sky is covered with clouds.
+            //
+            CLOUD_LAYER_LEVEL_2 = 4;
+
+            // 3/8 of the sky is covered with clouds.
+            //
+            CLOUD_LAYER_LEVEL_3 = 5;
+
+            // 4/8 of the sky is covered with clouds.
+            //
+            CLOUD_LAYER_LEVEL_4 = 6;
+
+            // 5/8 of the sky is covered with clouds.
+            //
+            CLOUD_LAYER_LEVEL_5 = 7;
+
+            // 6/8 of the sky is covered with clouds.
+            //
+            CLOUD_LAYER_LEVEL_6 = 8;
+
+            // 7/8 of the sky is covered with clouds.
+            //
+            CLOUD_LAYER_LEVEL_7 = 9;
+
+            // 8/8 of the sky is covered with clouds.
+            //
+            CLOUD_LAYER_LEVEL_8 = 10;
+
+            // Sky obscured, describes situations where the sky is not perceivable, e.g. in dense fog.
+            //
+            CLOUD_LAYER_LEVEL_SKY_OBSCURED = 11;
+        }
+    }
+
+    //
+    // \brief Defines wind properties.
+    //
+    message Wind
+    {
+        // The origin direction of the wind (not the target direction) in the ground/xy-plane of the \c
+        // world coordinate system. Corresponds to the heading/yaw angle, counted counterclockwise. \c
+        // 0 pointing north. If north not explicitly defined via proj4 reference in ground truth, \c
+        // then north is pointing in y-axis direction. The default orientation (x, y, z) is easting, northing, up [1].
+        //
+        // Unit: rad
+        //
+        // The preferred angular range is [0, 2pi].
+        //
+        // \note The direction is the origin, not the target direction.
+        //
+        // \par References:
+        // [1] https://proj.org/usage/projections.html#:~:text=E%20%20%20%200dN%200.000-,Axis%20orientation,%C2%B6,-Starting%20in%20PROJ
+        //
+        optional double origin_direction = 1;
+
+        // The wind speed.
+        //
+        // Unit: m/s
+        //
+        // Range [0...inf[
```

Comment:
[0..inf[ not really useful for the range. Maybe just state that velocity is >= 0, or integrate a rule like for the sun?